specutils/analysis/template_comparison.py

```diff
@@ -160,3 +183,54 @@ def template_match(observed_spectrum, spectral_templates,
             smallest_chi_index = index
 
     return smallest_chi_spec, chi2_min, smallest_chi_index
+
+
+def template_redshift(observed_spectrum, template_spectrum, min_redshift, max_redshift, delta_redshift):
+    """
+    Find the most accurate redshift for template_spectrum to match observed_spectrum using chi2
+
+    Parameters
+    ----------
+    observed_spectrum : :class:`~specutils.Spectrum1D`
+        The observed spectrum.
+    template_spectrum : :class:`~specutils.Spectrum1D`
+        The template spectrum, which will have it's redshift calculated
+    min_redshift : `float`
+        The minimum redshift allowed
+    max_redshift : `float`
+        The maximum redshift allowed
+    delta_redshift : `float`
+        The amount the redshift will change between loops
+
+    Returns
+    -------
+    final_redshift : `float`
+        The most accurate redshift for template_spectrum to match the observed_spectrum
+    redshifted_spectrum: :class:`~specutils.Spectrum1D`
+        A new Spectrum1D object which incorporates the template_spectrum with a spectral_axis
+        that has been redshifted using the final_redshift
+    """
+    if min_redshift > max_redshift:
+        raise ValueError("The `max_redshift` value must be greater than `min_redshift`.")
+        return
+
+    redshift = min_redshift
+    chi2_min = None
+    final_redshift = None
+
+    # Loop which goes through available redshift options and finds the smallest chi2
+    while redshift <= max_redshift:
+
+        # Create new redshifted spectrum and run it through the chi2 method
+        redshifted_spectrum = Spectrum1D(spectral_axis=template_spectrum.spectral_axis*(1+redshift),
```

Comment:
This doesn't pass through the uncertainty or `meta` or anything else like that.  Perhaps instead you should use `Spectrum1D(old_spectrum, spectral_axis=...)`?  That seems to do copying-while-still-resetting-the-spectral_axis...